model/app.model.js

```diff
@@ -32,15 +33,46 @@ function fetchAllArticles() {
 FROM
     articles
 LEFT JOIN 
-    comments ON comments.article_id = articles.article_id 
+    comments ON comments.article_id = articles.article_id
+WHERE
+	topic=$1
+GROUP BY 
+    articles.article_id
+ORDER BY 
+    articles.created_at DESC;`,
+				[query.topic]
+			)
+			.then((result) => {
+				if (result.rows.length === 0) {
+					return Promise.reject({ status: 404, msg: "Not found!" });
+				}
+				return result.rows;
+			});
+	} else {
+		return db
+			.query(
+				`SELECT 
+    articles.author,
+    articles.title,
+    articles.article_id,
+    articles.topic,
+    articles.created_at,
+    articles.votes,
+    articles.article_img_url,
+    COUNT(comments.article_id) AS comment_count 
+FROM
+    articles
+LEFT JOIN 
+    comments ON comments.article_id = articles.article_id
 GROUP BY 
     articles.article_id
 ORDER BY 
     articles.created_at DESC;`
-		)
-		.then((result) => {
-			return result.rows;
-		});
+			)
```

Comment:
I see here that we are doing one SQL query if there's a topic and another if there isn't. This will work no problem.

However, an alternative is query string concatenation (using +=). You may have seen it done this way. Potentially you could refactor to this if you have time 👍